tests/ui/binop/issue-77910-1.rs

```diff
@@ -7,5 +7,6 @@ fn main() {
     // we shouldn't ice with the bound var here.
     assert_eq!(foo, y);
     //~^ ERROR binary operation `==` cannot be applied to type
-    //~| ERROR `for<'a> fn(&'a i32) -> &'a i32 {foo}` doesn't implement `Debug`
+    //~| ERROR `for<'a> fn(&'a i32) -> &'a i32 {foo}: FnPtr` is not satisfied
```

Comment:
I personally think this error is slightly worse :thinking: slightly in favor of fast rejecting `FnDef` as well.